src/functions-and-arrays.js

```diff
@@ -1,19 +1,79 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(num1, num2) {
+  if(num1 >= num2){
+    return num1
+  } else {
+    return num2
+  }
+}
+
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+function findLongestWord(words){
+  let letters = wordsLength(words);
+  let longestIndex = 0;
+  if(words.length === 0){
+    return null
+  } else {
+    for(let i = 0; i < letters.length; i++){
+      if(letters[i] > letters[longestIndex]){
+        longestIndex = i;
+      }
+    }
+  }
+  return words[longestIndex];
+}
+
+function wordsLength(words){
+  let letters = [];
+  words.forEach(word => {
+    letters.push(word.length);
+  });
+  return letters;
+}
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+function sumNumbers(numbers){
+  let totalSum = 0;
+  for(let i = 0; i < numbers.length; i++){
+    totalSum += numbers[i];
+  }
+  return totalSum;
+}
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(numbers){
+  if(numbers.length === 0){
+    return null
+  } else {
+    let result = sumNumbers(numbers)/numbers.length;
+    return result;
+  }  
+}
+
+
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
+function averageWordLength(wordsArr){
+  if(wordsArr.length === 0){
+    return null
+  } else {
+    let OfwordsLength = wordsLength(wordsArr);
+    let average = sumNumbers(OfwordsLength)/wordsArr.length;
+    return average;  
+  }
+}
+
+
+
```

Comment:
there are missing variables in this function. Another solution might be to declare a variable with the value 0 then use a for loop to count the length of each word in the array. You can then return the total variable divided by the length of the number of elements in the array... Something like this: 

function averageWordLength(wordsArr) {
  if (wordsArr.length === 0) {
    return null;
  }
  let total = 0;
  for (let i = 0; i < wordsArr.length; i++) {
    total += wordsArr[i].length;
  }
  let average = total / wordsArr.length;

  return average;
}

Let me know if that makes sense!